packages/block-library/src/rss/index.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Server-side rendering of the `core/rss` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the rss error message.
+ *
+ * @param string $message RSS error message.
+ * @return string|void Void if renders on frontend.
+ */
+function render_rss_error_message( $message ) {
+	return '<div class="components-placeholder"><div class="notice notice-alt notice-error">' . $message . '</div></div>';
+}
+
+/**
+ * Renders the `core/rss` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the block content with received rss items.
+ */
+function render_block_core_rss( $attributes ) {
+
+	if ( ! isset( $attributes['feedURL'] ) ) {
+		return;
+	}
+
+	$url = filter_var( $attributes['feedURL'], FILTER_VALIDATE_URL );
+
+	// self-url destruction sequence.
+	if ( in_array( untrailingslashit( $url ), array( site_url(), home_url() ) ) ) {
+		return render_rss_error_message( __( 'Use \'Latest Posts\' block for this domain.', 'gutenberg' ) );
```

Comment:
I think this check/siggestion should only be in the editor, not in the render callback. I‘d remove it here.